docs/content/documentation/systems/AuxKernels/tensor_mechanics/RankTwoScalarAux.md

```diff
@@ -1,10 +1,232 @@
+
+### Second Invariant
+Similarly, the scalar type `SecondInvariant` finds the second invariant of the Rank-2 tensor, $T_{ij}$, as shown in Eq \eqref{eq:second_invarient_scalar_type}
+This method is defined in \cite{hjelmstad2007fundamentals}.
+\begin{equation}
+\label{eq:second_invarient_scalar_type}
+II_T = T_{ii} T_{jj} - \frac{1}{2} \left( T_{ij} T_{ij} + T_{ji} T_{ji} \right)
+\end{equation}
+
+#### Example Input File Syntax
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxKernels/si
+
+and an AuxVariable is required to store the AuxKernel information. Note that the name of the AuxVariable is used as the argument for the `variable` input parameter in the `RankTwoScalarAux` block.
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxVariables/secondinv
+
+### Third Invariant
+The scalar type `ThirdInvariant` computes the value of the Rank-2 tensor, $T_ij$, third invariant as given in Eq \eqref{eq:third_invarient_scalar_type} from \cite{malvern1969introduction}.
+\begin{equation}
+\label{eq:third_invarient_scalar_type}
+III_T = det \left( T_{ij} \right)  = \frac{1}{6} e_{ijk} e_{pqr} T_{ip} T_{jq} T_{kr}
+\end{equation}
+where $e$ is the Rank-3 permutation tensor.
+
+#### Example Input File Syntax
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxKernels/ti
+
+and an AuxVariable is required to store the AuxKernel information. Note that the name of the AuxVariable is used as the argument for the `variable` input parameter in the `RankTwoScalarAux` block.
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxVariables/thirdinv
+
+
+## L2 Norm
+The scalar type `L2Norm` calculates the L2 normal of a Rank-2 tensor, $T_{ij}$, as shown in Eq \eqref{eq:l2_norm_scalar_type}.
+\begin{equation}
+\label{eq:l2_norm_scalar_type}
+s = \sqrt{T_{ij} T_{ij}}
+\end{equation}
+
+## Principal Values
+### Maximum Principal Quantity
+The scalar type `MaxPrincipal` calculates the largest principal value for a symmetric tensor, using the calcEigenValues method from the Rank Two Tensor utility class.
+
+#### Example Input File Syntax
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxKernels/maxprincipal
+
+and an AuxVariable is required to store the AuxKernel information. Note that the name of the AuxVariable is used as the argument for the `variable` input parameter in the `RankTwoScalarAux` block.
+!listing modules/tensor_mechanics/test/tests/elastic_patch/elastic_patch.i block=AuxVariables/maxprincipal
+
+### Middle Principal Quantity
+Similarly, the scalar type `MidPrincipal` finds the second largest principal value for a symmetric tensor, using the calcEigenValues method from the Rank Two Tensor utility class.
+This method is valid only for 3D problems and will return an error if called in 2D or 1D simulations.
```

Comment:
I don't think that's true.  We compute the full 3D tensor regardless of the dimensionality of the problem.

Reply:
Yep--I tried to use midprincipal with a 2D problem and it worked just fine.  I updated the header file comments too